cartographer/io/serialization_format_migration.cc

```diff
@@ -18,284 +18,200 @@
+        proto.mutable_submap()->mutable_submap_id()->set_trajectory_id(
+            proto.submap().submap_id().trajectory_id());
+        submap_id_to_submap.Insert(
+            mapping::SubmapId{proto.submap().submap_id().trajectory_id(),
+                              proto.submap().submap_id().submap_index()},
+            proto.submap());
+        break;
+      }
+      case SerializedData::kNode: {
+        proto.mutable_node()->mutable_node_id()->set_trajectory_id(
+            proto.node().node_id().trajectory_id());
+        const mapping::NodeId node_id(proto.node().node_id().trajectory_id(),
+                                      proto.node().node_id().node_index());
+        const transform::Rigid3d& node_pose = node_poses.at(node_id);
+        pose_graph.AddNodeFromProto(node_pose, proto.node());
+        node_id_to_node.Insert(node_id, proto.node());
+        break;
+      }
+      case SerializedData::kTrajectoryData: {
+        proto.mutable_trajectory_data()->set_trajectory_id(
+            proto.trajectory_data().trajectory_id());
+        pose_graph.SetTrajectoryDataFromProto(proto.trajectory_data());
+        break;
+      }
+      case SerializedData::kImuData: {
+        pose_graph.AddImuData(proto.imu_data().trajectory_id(),
+                              sensor::FromProto(proto.imu_data().imu_data()));
+        break;
+      }
+      case SerializedData::kOdometryData: {
+        pose_graph.AddOdometryData(
+            proto.odometry_data().trajectory_id(),
+            sensor::FromProto(proto.odometry_data().odometry_data()));
+        break;
+      }
+      case SerializedData::kFixedFramePoseData: {
+        pose_graph.AddFixedFramePoseData(
+            proto.fixed_frame_pose_data().trajectory_id(),
+            sensor::FromProto(
+                proto.fixed_frame_pose_data().fixed_frame_pose_data()));
+        break;
+      }
+      case SerializedData::kLandmarkData: {
+        pose_graph.AddLandmarkData(
+            proto.landmark_data().trajectory_id(),
+            sensor::FromProto(proto.landmark_data().landmark_data()));
+        break;
+      }
+      default:
+        LOG(WARNING) << "Skipping unknown message type in stream: "
+                     << proto.GetTypeName();
+    }
+  }
 
-void SerializeToVersion1Format(
-    const ProtoMap& deserialized_data,
-    cartographer::io::ProtoStreamWriterInterface* const output) {
-  const std::vector<int> kFieldSerializationOrder = {
-      SerializedData::kPoseGraphFieldNumber,
-      SerializedData::kAllTrajectoryBuilderOptionsFieldNumber,
-      SerializedData::kSubmapFieldNumber,
-      SerializedData::kNodeFieldNumber,
-      SerializedData::kTrajectoryDataFieldNumber,
-      SerializedData::kImuDataFieldNumber,
-      SerializedData::kOdometryDataFieldNumber,
-      SerializedData::kFixedFramePoseDataFieldNumber,
-      SerializedData::kLandmarkDataFieldNumber};
+  // TODO(schwoere): Remove backwards compatibility once the pbstream format
```

Comment:
Nit: Will @schwoere actually ever work on this code or would it make sense to assign it to yourself?

Reply:
Style is to name the person with most context which is Martin in this case, not to assign them. See https://google.github.io/styleguide/cppguide.html#TODO_Comments. This also has the benefit that we do not have to speculate about the future. We could remove his name from the TODO but I do not see a benefit.